samples/basic/threads/src/main.c

```diff
@@ -48,23 +59,20 @@ void blink2(void)
 	gpio_pin_configure(gpiod, LED1, GPIO_DIR_OUT);
 	while (1) {
 		gpio_pin_write(gpiod, LED1, cnt % 2);
+
+		struct printk_data_t tx_data = { .led = 1, .cnt = cnt };
```

Comment:
This should be from a mem pool, else the fifo "next" will start pointing to itself without the k_sleep ?

Reply:
Sorry, I don't know what a mem pool is.

I think the code is valid, I used this documentation to write it:
http://docs.zephyrproject.org/kernel/data_passing/fifos.html#reading-from-a-fifo

Reply:
Can you elaborate on this:

" the fifo "next" will start pointing to itself without the k_sleep"

?

Reply:
Ah, I am sending the fifo a stack address, my bad, will fix.

Reply:
Fixed.

Reply:
Good catch, thank you for reviewing.